apps/dashboard/src/routes/(search-pages)/resource/[resource]/holders/+page.svelte

```diff
@@ -0,0 +1,109 @@
+<script lang="ts">
+  import Row from '@components/info-box/Row.svelte'
+  import Link from '@components/_base/link/Link.svelte'
+  import type { PageData } from './$types'
+  import LoadingSpinner from '@components/_base/button/loading-spinner/LoadingSpinner.svelte'
+  import { formatTokenValue } from '@utils'
+  import Pagination from '@components/_base/table/basic-table/Pagination.svelte'
+  import { gatewayApi } from '@stores'
+  import type { ResourceHoldersCollectionItem } from '@common/gateway-sdk'
+
+  export let data: PageData
+
+  let holders: ResourceHoldersCollectionItem[][] = []
+
+  data.holders.then(({ items }) => {
+    holders = [...holders, items]
+  })
+
+  let cursor: string | undefined
+
+  data.holders.then((holders) => (cursor = holders.next_cursor ?? undefined))
+
+  let currentPage = 0
+  let disabledNext = true
+
+  $: if (currentPage < holders.length - 1 || cursor) {
+    disabledNext = false
+  } else {
+    disabledNext = true
+  }
+
+  const loadNextPage = async () => {
+    if (currentPage < holders.length - 1) {
+      currentPage++
+      return
+    }
+
+    if (!cursor) return
+
+    if (currentPage === holders.length - 1) {
+      const items = await $gatewayApi!.extensions
+        .getResourceHolders(data.resource.address, cursor ?? undefined)
+        .then((holders) => {
+          cursor = holders.next_cursor ?? undefined
+          return holders.items
+        })
+      holders = [...holders, items]
+      currentPage++
+    } else {
+      currentPage++
+    }
+  }
+</script>
+
+<div class="card info-card">
+  {#await data.holders}
+    <div class="loading-container">
+      <div class="loading">
+        <LoadingSpinner />
+      </div>
+    </div>
+  {:then _}
+    {#each holders[currentPage] as holder}
+      <Row>
+        <div slot="left">
+          <Link
+            url="/account/{holder.holder_address}"
+            text={holder.holder_address}
+          />
+        </div>
+
+        <div slot="right">
+          {holder.type === 'FungibleResource'
+            ? formatTokenValue(holder.amount).displayValue
+            : holder.non_fungible_ids_count}
+        </div>
+      </Row>
+    {/each}
+
+    <div class="pagination">
+      <Pagination
```

Comment:
What do you think about using `<InfiniteScroll` component rather than `<Pagination`?

Reply:
I don't think we're using Pagination anywhere to be honest 🤔